syft/frameworks/torch/tensors/interpreters/precision.py

```diff
@@ -845,6 +845,31 @@ def linear(*args):
 
                 module.linear = linear
 
+                def dropout(input, p=0.5, training=True, inplace=False):
+                    """
+                    The dropout class calls functional dropout. Hence overloading functional dropout
+                    so that even works for dropout layer class.
+                    Ref:  https://stackoverflow.com/questions/54109617/implementing-dropout-from-scratch
+                    """
+                    if training:
+                        binomial = torch.distributions.binomial.Binomial(probs=1 - p)
+
+                        # we must convert the normal tensor to fixed precision before multiplication
+                        noise = (
+                            (
+                                binomial.sample(input.shape).type(torch.FloatTensor)
+                                * (1.0 / (1.0 - p))
+                            )
+                            .fix_prec()
```

Comment:
Here put `.fix_prec(**input.get_class_attributes())`  to get the same field & precission  than te input otherwise you won't be able to multiply them in the more general cases

Reply:
Done :+1: